hellohilt/src/main/java/com/airbnb/mvrx/hellohilt/di/HiltMavericksViewModelFactory.kt

```diff
@@ -0,0 +1,72 @@
+package com.airbnb.mvrx.hellohilt.di
+
+import com.airbnb.mvrx.MavericksState
+import com.airbnb.mvrx.MavericksViewModel
+import com.airbnb.mvrx.MavericksViewModelFactory
+import com.airbnb.mvrx.ViewModelContext
+import dagger.hilt.DefineComponent
+import dagger.hilt.EntryPoint
+import dagger.hilt.EntryPoints
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+
+/**
+ * To connect Mavericks ViewModel creation with Hilt's dependency injection, add the following Factory and companion object to your MavericksViewModel.
+ *
+ * Example:
+ *
+ * class MyViewModel @AssistedInject constructor(...): MavericksViewModel<MyState>(...) {
+ *
+ *   @AssistedInject.Factory
+ *   interface Factory : AssistedViewModelFactory<MyViewModel, MyState> {
+ *     ...
+ *   }
+ *
+ *   companion object : MavericksViewModelFactory<MyViewModel, MyState> by hiltMavericksViewModelFactory()
+ * }
+ */
+
+inline fun <reified VM : MavericksViewModel<S>, S : MavericksState> hiltMavericksViewModelFactory() = HiltMavericksViewModelFactory<VM, S>(VM::class.java)
+
+class HiltMavericksViewModelFactory<VM : MavericksViewModel<S>, S : MavericksState>(
+    private val viewModelClass: Class<out MavericksViewModel<S>>
+) : MavericksViewModelFactory<VM, S> {
```

Comment:
Might someone want to override `initialState` too? it seems like that could be a reasonable thing to do. To easily allow this we could have a nullable lambda argument to `hiltMavericksViewModelFactory` to provide the state

Reply:
@elihart You can delegate the factory interface to the function above and still override `createState`